pkg/model/iam/iam_builder.go

```diff
@@ -759,6 +752,152 @@ func addCalicoSrcDstCheckPermissions(p *Policy) {
 	)
 }
 
+func addNodeupPermissions(p *Policy, enableHookSupport bool) {
+	addCertIAMPolicies(p)
+	addKMSGenerateRandomPolicies(p)
+	addASLifecyclePolicies(p, enableHookSupport)
+	p.unconditionalAction.Insert(
+		"ec2:DescribeInstances", // aws.go
+	)
+}
+
+func addEtcdManagerPermissions(p *Policy) {
+	resource := stringorslice.Slice([]string{"*"})
+
+	p.Statement = append(p.Statement,
+		&Statement{
+			Effect: StatementEffectAllow,
+			Action: stringorslice.Slice([]string{
+				"ec2:DescribeVolumes", // aws.go
+			}),
+			Resource: resource,
+		},
+		&Statement{
+			Effect: StatementEffectAllow,
+			Action: stringorslice.Of(
+				"ec2:AttachVolume",
+			),
+			Resource: resource,
+			Condition: Condition{
+				"StringEquals": map[string]string{
+					"aws:ResourceTag/k8s.io/role/master": "1",
+				},
+			},
+		},
+	)
+
+}
+
+func AddLegacyCCMPermissions(p *Policy) {
+	p.unconditionalAction.Insert(
+		"ec2:CreateSecurityGroup",
+		"ec2:CreateTags",
+	)
+}
+
+func AddCCMPermissions(p *Policy, clusterName string, cloudRoutes bool) {
+	resource := stringorslice.Slice([]string{"*"})
+
+	p.unconditionalAction.Insert(
+		"autoscaling:DescribeAutoScalingGroups",
+		"autoscaling:DescribeLaunchConfigurations",
```

Comment:
I wonder if we still need this.

Reply:
I don't think so. I couldn't see CCM use this. It is mentioned here, but this is probably dated https://cloud-provider-aws.sigs.k8s.io/prerequisites.html